### PR TITLE
Universal installation script and .desktop files added

### PR DESCRIPTION
**Pull request by hashhar**
A python script to install VSCode which can run and install on all three (Linux (both x86 and x64), Windows, Darwin (apple)) platforms.

Also includes a .desktop file so that Linux users may be able to add VSCode to their Applications menu and docks.

Tested as working on Windows 10 Home, Pro, Windows 8.1 Home, Ubuntu 15.10 (GNOME and Unity), Kali Linux 2.0. No Apple devices available.

Apple's installation script is yet to be implemented since I don't have Apple OSX and don't know anything about how it handles installation.

PS: I hope I did nothing wrong. First code contribution to VSCode.

### CODE REVIEW

**Comment by hashhar** [comment]
I may need some help to get the script going for OSX. I will try writing some more code to handle the edge cases (observed using comments in the file TODO, BUG, ENHANCEMENT etc.) and read some documentation about OSX.

But it may take longer because I am in the middle of my semester exams here.

File: resources/universal_installation_script/VSCode.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Encoding=UTF-8
+Name=Visual Studio Code
+Comment=Code Editing. Redefined. Build and debug modern web and cloud applications.
+Exec=code
```
Comment (hashhar) [inline resources/universal_installation_script/VSCode.desktop]: Depends on the symlink made by the installation script.

**Comment by joaomoreno** [comment]
Hi @hashhar, thanks for the effort! Especially thanks for the `.desktop` file which makes Linux integration much better. I will certainly take it into account for future releases.

The universal installation script seems to be useful for yourself and possibly for others, but I'm not sure this repository is the place for it. How about you create a repository under your account and distribute it from there?

**Comment by hashhar** [comment]
Thanks for the feedback.

**Comment by hashhar** [comment]
Also what's the current plan on the auto-update channel for Linux if I might ask.

**Comment by joaomoreno** [comment]
Sure, we intend to be distributed in the various package managers across Linux distributions. Don't think we will have the automagical auto-update in-app like Windows and OS X.

**Comment by bgse** [comment]
@joaomoreno I think the auto-update is something that most Linux people will not want since updating is responsibility of the package manager.

For Fedora, there is already a COPR provided by community, see here:

https://github.com/Microsoft/vscode/issues/229

I'm pretty sure that if you reach out for help in the respective distro communities, there'll be more of these that can bridge the gap until you have the resources to provide 'official' packaging.

**Comment by joaomoreno** [comment]
@bgse Maybe you misunderstood me, that is exactly my point of view too.

**Pull request closed without merging**